### PR TITLE
Micro generator to ensure new file system

### PR DESCRIPTION
**Pull request by EnzoVieira**
Since we are now refactoring the file system for this project, I think that this would be very useful to ensure our new structure consistency. It also will help our team to save time when creating new Components or Layouts, preventing them from wasting their time trying to figure out the pattern.

I've used Plop that is a "micro-generator framework" and should save our team from [context switching](https://www.petrikainulainen.net/software-development/processes/the-cost-of-context-switching/) when dealing with others repo that use a different pattern

This lib works as a CLI, using `npm run generate`
Captura de Tela 2023-01-07 às 16 58 28

### CODE REVIEW

**Comment by EnzoVieira** [comment]
Some examples:

Captura de Tela 2023-01-07 às 17 06 13
Captura de Tela 2023-01-07 às 17 06 40

File: generator/plopfile.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    prompts: [
+      {
+        type: "input",
+        name: "name",
+        message: "Component name please",
+      },
+      {
+        type: "confirm",
+        name: "shouldAddStyles",
+        message: "Shoul add moduled styles?",
```
Comment (RuiL1904) [inline generator/plopfile.js:14]:
```suggestion
        message: "Should add moduled styles?",
```

File: generator/plopfile.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    prompts: [
+      {
+        type: "input",
+        name: "name",
+        message: "Layout name please (CamelCase)",
+      },
+      {
+        type: "confirm",
+        name: "shouldAddComponents",
+        message: "Shoul add components folder used only in this layout?",
```
Comment (RuiL1904) [inline generator/plopfile.js:56]:
```suggestion
        message: "Should add components folder used only in this layout?",
```

Comment (EnzoVieira) [reply generator/plopfile.js]: ops 😅
done ✅

**Review by RuiL1904 — DISMISSED** [review]
Great job 🚀

**Review by RuiL1904 — CHANGES_REQUESTED** [review]
Make sure "Code Quality" check is passing 🙏🏼

**Comment by feliciofilipe** [comment]
You probably want to merge this to: `ff/refactor-filesystem`, right @EnzoVieira ?

**Comment by EnzoVieira** [comment]
> You probably want to merge this to: `ff/refactor-filesystem`, right @EnzoVieira ?

I thought that since this feature does not affect the frontend itself, it could be already in main. but on second thought, maybe it's better to merge into your branch to bring this new feat just when the new file system is done 😅.

i'm gonna change that 🙏🏻

**Review by RuiL1904 — APPROVED** [review]

**Pull request merged**